sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/NotAssignedPrivateMember.cs

```diff
@@ -58,6 +58,7 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                 {
                     var namedType = (INamedTypeSymbol)c.Symbol;
                     if (!namedType.IsClassOrStruct() ||
+                        namedType.GetAttributes().Any(IsStructLayoutAttribute) ||

```

Comment:
I really like this approach ;)

Reply:
+1.
Could perhaps have a very short comment explaining why e.g. _// ignore interop types - see #xxx_